test/genOp.js

```diff
@@ -0,0 +1,278 @@
+// the changes the op generator will make.
+const clone = o => (typeof o === 'object' ? JSON.parse(JSON.stringify(o)) : o)
+
+const randomKey = obj => {
+  // this works on arrays too!
+  if (Array.isArray(obj)) {
+    return obj.length === 0 ? undefined : randomInt(obj.length)
+  } else {
+    const keys = Object.keys(obj)
+    return keys.length === 0 ? undefined : keys[randomInt(keys.length)]
+  }
+}
+
+const letters = 'abxyz'
+
+// Generate a random new key for a value in obj.
+const randomNewKey = obj => {
+  if (Array.isArray(obj)) {
+    return randomInt(obj.length + 1)
+  } else {
+    let key
+    while (true) {
+      // Mostly try just use a small letter
+      key =
+        randomInt(20) === 0 ? randomWord() : letters[randomInt(letters.length)]
+      if (obj[key] === undefined) {
+        break
+      }
+    }
+    return key
+  }
+}
+
+// Generate a random object
+const randomThing = () => {
+  switch (randomInt(7)) {
+    case 0:
+      return null
+    case 1:
+      return ''
+    case 2:
+      return randomWord()
+    case 3:
+      const obj = {}
+      for (
+        let i = 1, end = randomInt(2), asc = 1 <= end;
+        asc ? i <= end : i >= end;
+        asc ? i++ : i--
+      ) {
+        obj[randomNewKey(obj)] = randomThing()
+      }
+      return obj
+    case 4:
+      return __range__(1, randomInt(2), true).map(j => randomThing())
+    case 5:
+      return 0 // bias toward zeros to find accidental truthy checks
+    case 6:
+      return randomInt(50)
+  }
+}
+
+// Pick a random path to something in the object.
+const randomPath = data => {
+  if (data == null || typeof data !== 'object') {
+    return []
+  }
+
+  const path = []
+  while (randomReal() < 0.9 && data != null && typeof data === 'object') {
+    const key = randomKey(data)
+    if (key == null) {
+      break
+    }
+
+    path.push(key)
+    data = data[key]
+  }
+
+  return path
+}
+
+const randomWalkPick = (w, container) => {
+  const path = randomPath(container.data)
+  let parent = container
+  let key = 'data'
+
+  for (let p of Array.from(path)) {
+    parent = parent[key]
+    key = p
+    w.descend(key)
+  }
+  const operand = parent[key]
+  return [path, parent, key, operand]
+}
+
+// Returns [path, parent, key] if we can drop here,
+// or null if no drop is possible
+const randomWalkDrop = (w, container) => {
+  if (container.data === undefined) {
+    return [[], container, 'data']
+  } else if (typeof container.data !== 'object' || container.data === null) {
+    return null // Can't insert into a document that is a string or number
+  }
+
+  let [path, parent, key, operand] = randomWalkPick(w, container)
+  if (typeof operand === 'object' && operand !== null) {
+    parent = operand
+  } else {
+    w.ascend()
+    w.reset()
+    path.pop()
+  }
+  key = randomNewKey(parent)
+  return [path, parent, key]
+}
+
+const set = (container, key, value) =>
+  value === undefined
```

Comment:
By the way, I manually changed a coupld of these `if` `else` sections to use these ternary operators instead. Happy to switch them back if you'd prefer.